### PR TITLE
CORE-8: add API documentation to the endpoints in terrain.routes.fileio

### PR DESCRIPTION
**Pull request by slr71**

### CODE REVIEW

File: src/terrain/routes/schemas/fileio.clj [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  places the file stat information in the multipart params, which can then be returned as the response body to the
+  upload endpoint."
+  (upload/->Upload
+   (assoc FileStat
+     :filename     (describe String "The name of the file being uploaded")
+     :content-type (describe String "The MIME type of the file being uploaded"))))
+
+(s/defschema UrlUploadRequestBody
+  {:dest (describe NonBlankString "The destination directory for the uploaded file.")
+   :addr (describe NonBlankString "The URL to retrieve the file contents from.")})
```
Comment (psarando) [inline src/terrain/routes/schemas/fileio.clj]: The `urlupload` service function was looking for an `:address` key in the request body, but this is validating for an `:addr` key now.

File: src/terrain/routes/schemas/fileio.clj [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+(s/defschema FileUploadQueryParams
+  {:dest (describe NonBlankString "The destination directory for the uploaded file.")})
+
+(def DataStoreUpload
+  "Schema for a data store upload file parameter. The multipart request has to be processed in middleware because
+  the input stream for the file contents is closed before the body of the endpoint implementation is reached. We're
+  handling this by forwarding the request to the data-info service in custom middleware. The custom middleware then
+  places the file stat information in the multipart params, which can then be returned as the response body to the
+  upload endpoint."
+  (upload/->Upload
```
Comment (psarando) [inline src/terrain/routes/schemas/fileio.clj]: ✨

File: src/terrain/routes/schemas/fileio.clj (hunk 1 above)
Comment (slr71) [reply src/terrain/routes/schemas/fileio.clj]: Wow, I don't know where I got the `:addr` key from. Thanks for catching that.

**Comment by slr71** [comment]
Thanks for the review!

**Pull request merged**